xdr3/encode.go

```diff
@@ -293,7 +291,11 @@ func (enc *Encoder) EncodeFixedOpaque(v []byte) (int, error) {
 
 	// Write any padding if needed.
 	if pad > 0 {
-		b := make([]byte, pad)
+		// the maximum value of pad is 3, so the scratch buffer should be enough
+		b := enc.scratchBuf[:pad]
+		for i := 0; i < pad; i++ {
+			b[i] = 0x0
+		}
```

Comment:
💭 I wonder what bugs using a shared scratch buffer will open up. Would it impact performance to use a function to get at the buf which zeroed the buf everytime? Maybe not worth it.

Reply:
It would surely impact performance, but I don't know to what extent

Reply:
Concurrency problems came to mind when I added the buffer, but that would require multiple goroutines to share the same encoder/decoder, which I don't think makes sense at all.